libmscore/note.cpp

```diff
@@ -763,7 +763,7 @@ int Note::tpc() const
 QString Note::tpcUserName(bool explicitAccidental) const
       {
       QString pitchName = tpc2name(tpc(), NoteSpellingType::STANDARD, NoteCaseType::AUTO, explicitAccidental);
-      QString octaveName = QString::number(((epitch() + ottaveCapoFret()) / 12) - 1);
+      QString octaveName = QString::number(((epitch() + ottaveCapoFret() - int(tpc2alter(tpc()))) / 12) - 1);
```

Comment:
Shouldn't `int(tpc2alter(tpc()))` be a `static_cast<int>(tpc2alter(tpc()))` in order to conform to our coding rules?

Reply:
I don't think so. `int(...)` is used far more throughout the code than `static_cast<int>(...)`.

Reply:
OK. The coding rules do say to prefer `static_cast<>()` over C-style casts, but say nothing about function-style casts.

Reply:
`int(...)` **is** a C++ style cast, `(int)...` is C-style

Reply:
Yes, `int()` is a function-style cast that was introduced in C++. I was saying that our coding rules make no mention of function-style casts either way, but do say to use `static_cast<>()` in lieu of C-style casts.

I prefer `static_cast<>()` overall because it's easy to spot, it's easy to search for, and it's ugly and cumbersome to type. That last one is important because casts are a code smell and should be discouraged.

Reply:
whith that 'smell' you have a valid point

Reply:
See https://rules.sonarsource.com/cpp/RSPEC-871:

    C-style casts and functional notation casts are largely functionally equivalent. However, when they do not invoke a converting constructor, C-style casts are capable of performing dangerous conversions between unrelated types and of changing a variable's const-ness. Attempt to do these things with an explicit C++-style cast, and the compiler will catch the error. Use a C-style or functional notation cast, and it cannot.

    Additionally, C++-style casts are preferred because they are visually striking. The visual subtlety of a C-style or functional cast may mask that a cast has taken place, but a C++-style cast draws attention to itself, and makes the the programmer's intention explicit.

Reply:
I've added that to the coding style now ;-)